libdirections-onboard/src/test/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouterTest.kt

```diff
@@ -96,6 +107,81 @@ class MapboxOnboardRouterTest {
         verify { routerCallback.onResponse(any()) }
     }
 
+    @Test
+    fun checkCallbackCalledOnCancel() {
+        coEvery { navigator.getRoute(any()) } coAnswers {
+            onboardRouter.cancel()
+            routerResultSuccess
+        }
+
+        val latch = CountDownLatch(1)
+        val callback: Router.Callback = object : Router.Callback {
+            override fun onResponse(routes: List<DirectionsRoute>) {
+                Assert.fail()
+            }
+
+            override fun onFailure(throwable: Throwable) {
+                Assert.fail()
+            }
+
+            override fun onCanceled() {
+                latch.countDown()
+            }
+        }
+        onboardRouter.getRoute(routerOptions, callback)
+
+        if (!latch.await(5, TimeUnit.SECONDS)) {
+            Assert.fail("onCanceled not called")
+        }
+    }
+
+    @Test
+    fun checkCallbackCalledOnCancel2() = coroutineRule.runBlockingTest {
```

Comment:
Leaving a note here to remind ourselves to debate, decide and commit as a team what approach / style we want to take to test this kind of functionality - as currently we have 5 options apparently verifying the same. If we want to keep some of them, for clarity / understandability, we should change the name to reflect what we're testing and how (instead of adding `2`, `3`, etc.). Will cut a `TODO` ticket so we don't forget. cc @LukasPaczos @korshaknn

Reply:
They are not verifying the same @Guardiola31337. Those tests are all verifying different scenarios where the `onCanceled` should be invoked.

Reply:
I believe that `checkCallbackCalledOnCancel` and `checkCallbackCalledOnCancel3` are testing the same but using a different approach. But yeah, they don't hurt as they're Unit tests and run fast. In any case, let's make it explicit in the test name 👍 which would also help avoiding confusion when 👀 😅

Reply:
Yea, we could've been more creative with names, let's fix it up as we go :)